dotnet/src/dotnetcore/Providers/Messaging/GXAzureEventGrid/AzureEventGrid.cs

```diff
@@ -50,22 +61,15 @@ public bool SendEvent(GXCloudEvent gxCloudEvent, bool binaryData)
 		{
 			CloudEvent evt = ToCloudEvent(gxCloudEvent, binaryData);
 			bool success = false;
-			try
+			Task<bool> task;
+			if (_client != null)
 			{
-				Task task;
-				if (_client != null)
-				{
-					task = Task.Run(async () => await sendEvtAsync(evt).ConfigureAwait(false));
-					success = true;
-				}
-				else
-				{
-					throw new Exception("There was an error at the Event Grid initialization.");
-				}
+				task = Task.Run(async () => await sendEvtAsync(evt).ConfigureAwait(false));
+				success = task.Result;
 			}
-			catch (AggregateException ae)
+			else
 			{
-				throw ae;
+				throw new Exception("There was an error at the Event Grid initialization.");
```

Comment:
Throwing exceptions is more expensive than setting an error code or returning a message. Have we already considered managing errors in a way other than throwing exceptions?